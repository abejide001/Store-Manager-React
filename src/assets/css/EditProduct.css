@import url("https://fonts.googleapis.com/css?family=Charmonman|Roboto|K2D");
#edit {
  min-height: 400px;
  font-family: "K2D", sans-serif;
  flex: 1;
}
.edit-showcase {
  display:  flex;
  flex-direction: column;
  height: 100%;
}
.edit-body {
  position: absolute;
  width: 100%;
  top:0;
  bottom: 0;
}
#edit h1 {
  font-size: 40px;
  text-align: center;
}

.submission-form {
  max-width: 500px;
  margin: 2rem auto;
  padding: 2rem;
  border-radius: 5px;
  margin-bottom: 100px;
}

.submission-form input {
  display: block;
  width: 100%;
  border: 2px solid gray;
  padding: 0.5rem;
  font-size: 18px;
  border-radius: 5px;
  margin: 10px;
}

#sendBtn {
  border: 0;
  background: gray;
  padding: 0.5rem;
  color: white;
  width: auto;
  cursor: pointer;
}