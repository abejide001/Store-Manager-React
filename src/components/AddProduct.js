/* eslint-disable camelcase */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { addProducts } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/AddProduct.css';

export class AddProduct extends Component {
  state = {
    name: '',
    price: '',
    quantity_in_inventory: '',
    product_image: '',
  };

  handleSubmit = (e) => {
    e.preventDefault();
    const {
      name, price, quantity_in_inventory, product_image,
    } = this.state;
    const productData = {
      name,
      price,
      quantity_in_inventory,
      product_image,
    };
    this.props.addProducts(productData);
  };

  handleChange = (e) => {
    this.setState({ [e.currentTarget.name]: e.currentTarget.value });
  };

  render() {
    const {
      name, price, quantity_in_inventory, product_image,
    } = this.state;
    return (
      <div>
        <header>
          <div className="container">
            <div id="header-title">
              <Link to="/">
                <img src={logo} alt="logo" />
              </Link>
            </div>
            <nav>
              <ul>
                <li>
                  <Link to="/admin">Admin</Link>
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <section id="create-product">
          <form className="submission-form" onSubmit={this.handleSubmit}>
            <h1>Create Product</h1>
            <input
              type="text"
              name="name"
              placeholder="Product Name"
              value={name}
              onChange={this.handleChange}
              required
            />
            <input
              type="number"
              name="price"
              placeholder="Price"
              onChange={this.handleChange}
              value={price}
              required
            />
            <input
              type="number"
              name="quantity_in_inventory"
              value={quantity_in_inventory}
              onChange={this.handleChange}
              placeholder="quantity in inventory"
              required
            />
            <input
              type="url"
              name="product_image"
              value={product_image}
              placeholder="Image URL"
              onChange={this.handleChange}
              required
            />
            <input type="submit" value="CREATE PRODUCT" id="sendBtn" />
          </form>
        </section>
      </div>
    );
  }
}
AddProduct.propTypes = {
  addProducts: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(mapStateToProps, { addProducts })(AddProduct);
