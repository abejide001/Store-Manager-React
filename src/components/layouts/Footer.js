import React from 'react';
import '../../assets/css/style.css';

export default function Footer() {
  return (
    <div>
      <footer>
        <div className="container">
          <h3>
            Store Manager
            {' '}
            {new Date().getFullYear()}
          </h3>
        </div>
      </footer>
    </div>
  );
}
