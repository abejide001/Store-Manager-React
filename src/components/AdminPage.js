/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts, deleteProduct } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/Admin.css';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Spinner from '../common/Spinner';

export class Admin extends Component {
  componentWillMount() {
    this.props.getProducts();
  }

handleDelete = (id) => {
  this.props.deleteProduct(id);
}

render() {
  const { products } = this.props.product.products;
  return (
    <div>
      <header>
        <div className="container">
          <div id="header-title">
            <Link to="/"><img src={logo} alt="logo" /></Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link to="/sale-records">Records</Link>
              </li>
              <li>
                <Link to="/create-product">Create Product</Link>
              </li>
              <li>
                <Link to="/available-products.html">Available Products</Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section id="product-area">
        <div className="container">
          <h1 id="product-header">Products</h1>
          <div className="products">
            {products === undefined ? <Spinner />
              : products.value.map(item => (
                <div className="product-item" key={item.id}>
                  <img src={item.product_image} className="product-image" />
                  <p className="product-name">{item.name}</p>
                  <p className="product-price">
#
                    {item.price}
                  </p>
                  <p>
                    <button className="editBtn"><Link to={`edit-product?id=${item.id}`}>Edit</Link></button>
                    {' '}
                    <button className="deleteBtn" type="submit" onClick={() => this.handleDelete(item.id)}>Delete</button>
                  </p>
                </div>
              ))}
          </div>
        </div>
      </section>
      <Footer />
    </div>
  );
}
}

const mapStateToProps = state => ({
  product: state.product,
  error: state.error,
});
export default connect(mapStateToProps, { getProducts, deleteProduct })(Admin);
