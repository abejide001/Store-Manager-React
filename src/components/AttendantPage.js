import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts } from '../actions/productActions';
import Footer from './layouts/Footer';
import logo from '../assets/images/logo.png';
import '../assets/css/AttendantPage.css';
import '../assets/css/style.css';
import Spinner from '../common/Spinner';

export class AttendantPage extends Component {
  state = {
    search: '',
  };

  componentDidMount() {
    this.props.getProducts();
  }

  handleChange = (e) => {
    this.setState({ search: e.target.value });
  };

  render() {
    const { products } = this.props.product.products;
    const { search } = this.state;
    const foundProducts = products
      && products.value.filter(
        product => product.name.toLowerCase().includes(search.toLowerCase()),
      );

    return (
      <div>
        <header>
          <div className="container">
            <div id="header-title">
              <Link to="/">
                <img src={logo} alt="logo" />
              </Link>
            </div>
            <nav>
              <ul>
                <li>
                  <Link to="./available-products">Available Products</Link>
                </li>
                <li>
                  <Link to="./individual-attendant-sale-record">Records</Link>
                </li>
                <li>
                  <input
                    type="search"
                    placeholder="Search Product"
                    value={search}
                    name="search"
                    onChange={this.handleChange}
                  />
                </li>
              </ul>
            </nav>
          </div>
        </header>
        <section id="product-area">
          <div className="container">
            <h2>Products</h2>
            <div className="products">
              {foundProducts === undefined ? (
                <Spinner />
              ) : (
                foundProducts.map(item => (
                  <div className="product-item" key={item.id}>
                    <img src={item.product_image} className="product-image" />
                    <p className="product-name">{item.name}</p>
                    <p className="product-price">
#
                      {(item.price).toLocaleString()}
                                        </p>
                    <p>
                      <button className="product-button" type="button">
                        ADD TO CART
                      </button>
                    </p>
                  </div>
                ))
              )}
            </div>
          </div>
        </section>
        <section className="container cart-section">
          <div className="container">
            <h1 id="cart-title">CART</h1>
            <div className="cart-row">
              <span className="cart-item cart-header cart-column">ITEM</span>
              <span className="cart-price cart-header cart-column">PRICE</span>
              <span className="cart-quantity cart-header cart-column">
                QUANTITY
              </span>
            </div>
            <div className="cart-items" />
            <div className="cart-total">
              <strong className="cart-total-title">Total</strong>
              <span className="cart-total-price">#0</span>
            </div>
            <button className="purchase-button" type="button">
              PURCHASE
            </button>
          </div>
        </section>
        <Footer />
      </div>
    );
  }
}
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(
  mapStateToProps,
  { getProducts },
)(AttendantPage);
