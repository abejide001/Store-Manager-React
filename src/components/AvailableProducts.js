/* eslint-disable jsx-a11y/alt-text */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getProducts } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/AvailableProducts.css';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Spinner from '../common/Spinner';

export class Admin extends Component {
  componentWillMount() {
    this.props.getProducts();
  }

  render() {
    const { products } = this.props.product.products;
    return (
    <div>
      <header>
        <div className="container">
          <div id="header-title">
            <Link to="/"><img src={logo} alt="logo" /></Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link to="/sale-records">Records</Link>
              </li>
              <li>
                <Link to="/add-product">Create Product</Link>
              </li>
              <li>
                <Link to="/available-products.html">Available Products</Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section id="product-area">
        <div className="container">
          <h1 id="product-header">Available Products</h1>
          <div className="products">
            {products === undefined ? <Spinner />
              : products.value.map(item => (
                <div className="product-item" key={item.id}>
                  <img src={item.product_image} className="product-image" />
                  <p className="product-name">{item.name}</p>
                  <p className="product-price">
#
                    {item.price}
                  </p>
                  <p>
                    {' '}
                    <button className="modalBtn" type="submit">Details</button>
                  </p>
                </div>
              ))}
          </div>
        </div>
      </section>
      <Footer />
    </div>
  );
}
}

const mapStateToProps = state => ({
  product: state.product,
  error: state.error,
});
export default connect(mapStateToProps, { getProducts })(Admin);
