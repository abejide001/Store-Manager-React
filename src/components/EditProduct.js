/* eslint-disable react/forbid-prop-types */
/* eslint-disable no-undef */
/* eslint-disable camelcase */
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Footer from './layouts/Footer';
import { editProduct } from '../actions/productActions';
import logo from '../assets/images/logo.png';
import '../assets/css/EditProduct.css';

export class EditProduct extends Component {
    state = {
      name: '',
      price: '',
      quantity_in_inventory: '',
      product_image: '',
    }

    handleSubmit = (event) => {
      event.preventDefault();
      const regex = /\d+/g;
      const id = this.props.location.search.match(regex);
      const { name, price, quantity_in_inventory, product_image } = this.state;
      const productData = {
        name,
        price,
        quantity_in_inventory,
        product_image,
      };
      this.props.editProduct(id, productData);
    }

    handleChange = (e) => {
      this.setState({ [e.currentTarget.name]: e.currentTarget.value });
    };

    render() {
      const {
        name, price, quantity_in_inventory, product_image,
      } = this.state;
      return (
        <div>
          <div className="edit-body">
            <div className="edit-showcase">
              <header>
                <div className="container">
                  <div id="header-title">
                    <Link to="/">
                      <img src={logo} alt="logo" />
                    </Link>
                  </div>
                  <nav>
                    <ul>
                      <li>
                        <Link to="/admin">Admin</Link>
                      </li>
                    </ul>
                  </nav>
                </div>
              </header>
              <section id="edit">
                <form className="submission-form" onSubmit={this.handleSubmit}>
                  <h1>EDIT PRODUCT</h1>
                  <input type="text" className="input1" name="name" placeholder="Product Name" value={name} onChange={this.handleChange} required />
                  <input type="number" className="input2" name="price" placeholder="Price" value={price} onChange={this.handleChange} required />
                  <input type="number" className="input3" name="quantity_in_inventory" placeholder="quanity in inventory" value={quantity_in_inventory} onChange={this.handleChange} required />
                  <input type="url" name="product_image" placeholder="Image URL" value={product_image} onChange={this.handleChange} required />
                  <input type="submit" value="EDIT" id="sendBtn" />
                </form>
              </section>
              <Footer />
            </div>
          </div>
        </div>
      );
    }
}
EditProduct.propTypes = {
  location: PropTypes.object.isRequired,
  editProduct: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  product: state.product,
});
export default connect(mapStateToProps, { editProduct })(EditProduct);
