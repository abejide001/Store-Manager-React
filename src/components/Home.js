import React from 'react';
import '../assets/css/style.css';
import Footer from './layouts/Footer';
import Header from './layouts/Header';
import HomeBody from './layouts/HomeBody';

export default function Home() {
  return (
    <div className="landing-page">
      <Header />
      <HomeBody />
      <Footer />
    </div>
  );
}
