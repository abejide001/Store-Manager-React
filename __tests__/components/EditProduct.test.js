/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router-dom';
import { EditProduct } from '../../src/components/EditProduct';
import Footer from '../../src/components/layouts/Footer';

let wrapper;
describe('<EditProduct />', () => {
  describe('render()', () => {
    beforeEach(() => {
      wrapper = shallow(<EditProduct />);
    });
    it('renders the component', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe('', () => {
    beforeEach(() => {
      const props = {
        product: null,
        editProduct: jest.fn(),
        location: {
          search: {
            match: jest.fn()
          }
        }
      }
      wrapper = mount(
        <MemoryRouter><EditProduct {...props} /></MemoryRouter>
      );
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should have a footer component', () => {
      expect(wrapper.find(Footer).length).toEqual(1);
    });
    it('should have one form element', () => {
      expect(wrapper.find('form').length).toEqual(1);
    });
    it('should have three input element', () => {
      expect(wrapper.find('input').length).toEqual(5);
    });
    it('should have two link element', () => {
      expect(wrapper.find('Link').length).toBe(2);
    });
    it('has an input with a name prop', () => {
      wrapper.find('.input1').simulate('change', {
        target: name,
      });
      wrapper.update();
      expect(wrapper.find('.input1').prop('name')).toEqual('name')
    })
    it('has an input with a name prop', () => {
      wrapper.find('.input2').simulate('change', {
        target: name,
      });
      wrapper.update();
      expect(wrapper.find('.input2').prop('name')).toEqual('price')
    })
    it('has an input with a name prop', () => {
      wrapper.find('.input3').simulate('change', {
        target: name,
      });
      wrapper.update();
      expect(wrapper.find('.input3').prop('name')).toEqual('quantity_in_inventory')
    })
    it('should empty the form on submit', () => {
      wrapper.find('form').simulate('submit');
      wrapper.update();
      expect(wrapper.find('.input1').prop('value')).toEqual('');
    });
  });
});
