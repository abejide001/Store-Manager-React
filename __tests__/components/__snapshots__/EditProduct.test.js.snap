// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProduct /> render() renders the component 1`] = `
<div>
  <div
    className="edit-body"
  >
    <div
      className="edit-showcase"
    >
      <header>
        <div
          className="container"
        >
          <div
            id="header-title"
          >
            <Link
              replace={false}
              to="/"
            >
              <img
                alt="logo"
                src="test-file-stub"
              />
            </Link>
          </div>
          <nav>
            <ul>
              <li>
                <Link
                  replace={false}
                  to="/admin"
                >
                  Admin
                </Link>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <section
        id="edit"
      >
        <form
          className="submission-form"
          onSubmit={[Function]}
        >
          <h1>
            EDIT PRODUCT
          </h1>
          <input
            className="input1"
            name="name"
            onChange={[Function]}
            placeholder="Product Name"
            required={true}
            type="text"
            value=""
          />
          <input
            className="input2"
            name="price"
            onChange={[Function]}
            placeholder="Price"
            required={true}
            type="number"
            value=""
          />
          <input
            className="input3"
            name="quantity_in_inventory"
            onChange={[Function]}
            placeholder="quanity in inventory"
            required={true}
            type="number"
            value=""
          />
          <input
            name="product_image"
            onChange={[Function]}
            placeholder="Image URL"
            required={true}
            type="url"
            value=""
          />
          <input
            id="sendBtn"
            type="submit"
            value="EDIT"
          />
        </form>
      </section>
      <Footer />
    </div>
  </div>
</div>
`;
